JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/categories/CategoriesScreen.kt

```diff
@@ -70,10 +71,8 @@ fun CategoriesScreen(
                     tvLazyGridState.firstVisibleItemScrollOffset < 100
         }
     }
-    val movieRepository = LocalMovieRepository.current!!
-    val movieCategories = remember {
-        movieRepository.getMovieCategories()
-    }
+    val categoriesScreenViewModel: CategoriesScreenViewModel = viewModel()
+    val movieCategories by categoriesScreenViewModel.categoryList.collectAsState()
```

Comment:
Use [`collectAsStateWithLifecycle`](https://developer.android.com/reference/kotlin/androidx/lifecycle/compose/package-summary#(kotlinx.coroutines.flow.Flow).collectAsStateWithLifecycle(kotlin.Any,androidx.lifecycle.Lifecycle,androidx.lifecycle.Lifecycle.State,kotlin.coroutines.CoroutineContext)) everywhere.

Reply:
Thanks for the suggestion, and I am bit confused with it. The ViewModels calls stateIn method over the flows as recommended in the post [1]. It looks an alternative way to call collectAsStatewithLifecycle in view codes. Would you please let me if I am misunderstanding the recommendation, or [1] is outdated.

[1] https://medium.com/androiddevelopers/consuming-flows-safely-in-jetpack-compose-cde014d0d5a3.